indieauth.php

```diff
@@ -18,9 +18,55 @@
 	define( 'INDIEAUTH_REMOTE_MODE', 0 );
 }
 
+register_activation_hook( __FILE__, array( 'IndieAuth_Plugin', 'schedule' ) );
+register_deactivation_hook( __FILE__, array( 'IndieAuth_Plugin', 'deactivation' ) );
+
+
+add_action( 'upgrader_process_complete', array( 'IndieAuth_Plugin', 'upgrader_process_complete' ), 10, 2 );
+add_action( 'indieauth_cleanup', array( 'IndieAuth_Plugin', 'expires' ) );
+
 class IndieAuth_Plugin {
 	public static $indieauth = null; // Loaded instance of authorize class
 
+	/*
+	 * Process to Trigger on Plugin Update.
+	 */
+	public static function upgrader_process_complete( $upgrade_object, $options ) {
+		$current_plugin_path_name = plugin_basename( __FILE__ );
+		if ( ( 'update' === $options['action'] ) && ( 'plugin' === $options['type'] ) ) {
+			foreach ( $options['plugins'] as $each_plugin ) {
+				if ( $each_plugin === $current_plugin_path_name ) {
+					self::schedule();
+				}
+			}
+		}
+	}
+
+	public static function deactivation() {
+		$timestamp = wp_next_scheduled( 'indieauth_cleanup', array( false ) );
+		if ( $timestamp ) {
+			wp_unschedule_event( $timestamp, 'indieauth_cleanup', array( false ) );
+		}
+	}
+
+	public static function schedule() {
```

Comment:
rename to `activation`?

Reply:
But it isn't just called on activation. I can create a function that calls it called activation

Reply:
Done now.

Reply:
ah, you added a new  function instead of renaming this one! nice 👍